src/main/scala/chisel3/util/Inline.scala

```diff
@@ -0,0 +1,96 @@
+// See LICENSE for license details.
+
+package chisel3.util
+
+import chisel3._
+import chisel3.internal.InstanceId
+import chisel3.experimental.{BaseModule, ChiselAnnotation, RunFirrtlTransform}
+import firrtl.passes.{InlineAnnotation, InlineInstances}
+import firrtl.transforms.NoDedupAnnotation
+import firrtl.annotations.{CircuitName, ModuleName, ComponentName}
+
+/** Inlines an instance of a [[chisel3.experimental.BaseModule]]
+  *
+  * @example {{{
+  *  /* Provides some dummy module internals */
+  *  trait Internals { this: Module =>
+  *    val io = IO(new Bundle{ val a = Input(Bool()) })
+  *  }
+  *  /* InlineInstance is mixed directly into Foo's definition. Every instance
+  *   * of this will be inlined. */
+  *  class Foo extends Module with Internals with InlineInstance
+  *  /* Bar will, by default, not be inlined */
+  *  class Bar extends Module with Internals
+  *  class Top extends Module {
+  *    val io = IO(new Bundle{ val a = Input(Bool()) })
+  *    val x = Module(new Foo)                     // x will be inlined
+  *    val y = Module(new Bar with InlineInstance) // y will also be inlined
+  *    val z = Module(new Bar)                     // z will not be inlined
+  *    Seq(x, y, z).map(_.io.a := io.a)
+  *  }
+  * }}}
+  */
+trait InlineInstance { self: BaseModule =>
```

Comment:
I think the trait is useful to inline all instances of a given Module, do you think we could also have a way to inline a single instance?

Reply:
Oh, I see you do it in the example! Nice :)